go-controller/pkg/ovn/gateway.go

```diff
@@ -533,16 +533,22 @@ func (gw *GatewayManager) GatewayInit(
 			return item.IPPrefix == lrsr.IPPrefix &&
 				libovsdbops.PolicyEqualPredicate(lrsr.Policy, item.Policy)
 		}
-		err := libovsdbops.CreateOrReplaceLogicalRouterStaticRouteWithPredicate(gw.nbClient,
-			gw.clusterRouterName, &lrsr, p, &lrsr.Nexthop)
-		if err != nil {
-			return fmt.Errorf("error creating static route %+v in %s: %v", lrsr, gw.clusterRouterName, err)
+
+		if gw.clusterRouterName != "" {
```

Comment:
These routes are our:
```
100.64.0.1/32 via 100.64.0.1 dst-ip
100.64.0.2/32 via 100.64.0.2 dst-ip
```
which we use to override the source ip route that only points to one GR:

```
10.244.0.0/24 via 100.64.0.1 src-ip
```

This was used when we have multiple GRs connected to a single ovn_cluster_router (non-IC). In IC we do:

```
               100.64.0.2                100.64.0.2 dst-ip
               100.64.0.3                100.88.0.3 dst-ip
```

Where we send it to the transit IP of the other node. These routes protect us for if the packet comes from another GR on another node with the source IP of the 100.64.0.x (join IP). Then on the reply, we want the packet to go back to the right GR and not just use the source ip route to send it to our *own* GR.

With layer 2 network there is no ovn_cluster_router, so the reply packet will always go back to the right GR I think.

Reply:
Right no action to be taken. I just wanted to make note of this so I dont forget in the future why its not there for layer 2.